src/couch_replicator/src/couch_replicator_auth_session.erl

```diff
@@ -288,7 +288,9 @@ extract_creds_from_url(Url) ->
             Prefix = lists:concat([Proto, "://", User, ":", Pass, "@"]),
             Suffix = lists:sublist(Url, length(Prefix) + 1, length(Url) + 1),
             NoCreds = lists:concat([Proto, "://", Suffix]),
-            {ok, User, Pass, NoCreds}
+            User1 = mochiweb_util:unquote(User),
+            Pass1 = mochiweb_util:unquote(Pass),
```

Comment:
I think we should call [`chttpd:unquote/1`](https://github.com/apache/couchdb/blob/main/src/chttpd/src/chttpd.erl#L678) here instead. This would make moving away from mochiweb framework easier in the future.

Reply:
There are pros and cons to either way. For example `chttpd` is not mentioned as a dependency in the app file. However we do call functions from `chttpd`. I am fine with leaving it as is.

Reply:
Thanks for reviewing my PR @iilyak, I replaced the quoted line... and waiting for the tests to pass now.